scene/3d/skeleton_3d.cpp

```diff
@@ -369,67 +369,82 @@ void Skeleton3D::_notification(int p_what) {
 	}
 }
 
-void Skeleton3D::clear_bones_global_pose_override() {
-	for (int i = 0; i < bones.size(); i += 1) {
-		bones.write[i].global_pose_override_amount = 0;
-		bones.write[i].global_pose_override_reset = true;
+void Skeleton3D::clear_bones_pose_override(bool p_reset) {
+	const int bone_size = bones.size();
+	for (int i = 0; i < bone_size; i += 1) {
+		bones.write[i].is_override_enabled = false;
+		bones.write[i].pose_cache_dirty = true;
+	}
+	if (p_reset) {
+		for (int i = 0; i < bone_size; i += 1) {
+			bones.write[i].pose_override = Transform3D();
+		}
 	}
 	_make_dirty();
 }
 
-void Skeleton3D::set_bone_global_pose_override(int p_bone, const Transform3D &p_pose, real_t p_amount, bool p_persistent) {
+void Skeleton3D::set_bone_pose_override_enabled(int p_bone, bool p_enabled, bool p_reset) {
 	const int bone_size = bones.size();
 	ERR_FAIL_INDEX(p_bone, bone_size);
-	bones.write[p_bone].global_pose_override_amount = p_amount;
-	bones.write[p_bone].global_pose_override = p_pose;
-	bones.write[p_bone].global_pose_override_reset = !p_persistent;
+	bones.write[p_bone].is_override_enabled = p_enabled;
+	if (p_reset) {
+		bones.write[p_bone].pose_override = Transform3D();
+	}
+	bones.write[p_bone].pose_cache_dirty = true;
 	_make_dirty();
-}
+};
 
-Transform3D Skeleton3D::get_bone_global_pose_override(int p_bone) const {
+bool Skeleton3D::get_bone_pose_override_enabled(int p_bone) const {
```

Comment:
Since it's a boolean getter, `is_bone_global_pose_override_enabled` might be better.